File: src-1.x/BYTES.NET/Imaging/API/ImageParserMetadata.vb
'import internal namespace(s) required
Imports BYTES.NET.MEF.API

Namespace Imaging.API

    Public Class ImageParserMetadata

        Inherits Metadata

#Region "public properties"

        Public Property Name As String

        Public Property FileExtensions As String()

#End Region

#Region "public new instance method(s)"

        ''' <summary>
        ''' default new instance method
        ''' </summary>
        Public Sub New()

            MyBase.New(GetType(IImageParser))

        End Sub

#End Region

    End Class

End Namespace

File: src-2023/BYTES.NET.Math/Imaging/API/ImageParserMetadata.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BYTES.NET.Math.Imaging.API
{
    /// <summary>
    /// Represents metadata for an image parser, extending the Metadata class.
    /// </summary>
    public class ImageParserMetadata
    {
        #region Public Properties

        /// <summary>
        /// Gets or sets the name of the image parser.
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets the supported file extensions for the image parser.
        /// </summary>
        public string[] FileExtensions { get; set; }

        #endregion

        #region Public New Instance Method(s)

        /// <summary>
        /// Default constructor for ImageParserMetadata.
        /// </summary>
        public ImageParserMetadata()
        {
        }

        #endregion
    }
}
